ui/pages/Project/selectors.js

```diff
@@ -248,51 +283,37 @@ export const getVisibleFamilies = createSelector(
 export const getVisibleFamiliesInSortedOrder = createSelector(
   getVisibleFamilies,
   getIndividualsByGuid,
-  getSamplesByGuid,
+  getSamplesByFamily,
   getFamiliesSortOrder,
   getFamiliesSortDirection,
-  (visibleFamilies, individualsByGuid, samplesByGuid, familiesSortOrder, familiesSortDirection) => {
+  (visibleFamilies, individualsByGuid, samplesByFamily, familiesSortOrder, familiesSortDirection) => {
     if (!familiesSortOrder || !FAMILY_SORT_LOOKUP[familiesSortOrder]) {
       return visibleFamilies
     }
 
-    const getSortKey = FAMILY_SORT_LOOKUP[familiesSortOrder](individualsByGuid, samplesByGuid)
-
-    return orderBy(visibleFamilies, [getSortKey], [familiesSortDirection > 0 ? 'asc' : 'desc'])
+    const getSortKey = FAMILY_SORT_LOOKUP[familiesSortOrder](individualsByGuid, samplesByFamily)
+    return visibleFamilies.slice(0).sort((a, b) => getSortKey(a).localeCompare(getSortKey(b)) * familiesSortDirection)
   },
 )
 
-export const getEntityExportConfig = (project, rawData, tableName, fileName, fields) => ({
+export const getEntityExportConfig = ({ project, tableName, fileName, fields }) => ({
```

Comment:
I have a question about why the object parameter is used rather than separate ones.  Is it just for optional parameters? How about using the default parameters like Python?

Reply:
because remembering an exact argument order is hard, especially when not all the fields are required and some calls to this function only provide some of them. Plus, using the object you kind of get to name the arguments which makes it more readable. So for instance with this change 2 different calls would look like 
```
getEntityExportConfig({ project, tableName: 'project_families', fileName: 'families', fields: FAMILY_EXPORT_DATA })
...
getEntityExportConfig({ project, fileName: 'template', fields })
```
Whereas using the separate parameters that would read like
```
getEntityExportConfig(project, 'project_families', 'families', FAMILY_EXPORT_DATA)
...
getEntityExportConfig(project, null, 'template', fields)
```
I just find it much more readable/ maintainable the way it is now